plugins/app/src/hooks/useDynamicScript.ts

```diff
@@ -1,21 +1,49 @@
 import { useEffect, useState } from 'react';
 
-const loadedScripts: { [name: string]: Promise<void> } = {};
-
-export const useDynamicScript = (
-  name: string,
-  version: string,
-): {
-  failed: boolean;
-  ready: boolean;
-} => {
+const loadedScripts: Record<string, Promise<void>> = {};
+
+export type ScriptStatus = { isFailed: boolean; isReady: boolean };
+
+/**
+ * Load a kobs specific federated module from a remote server.
+ * @param name the plugin name aka. federated module name
+ * @param version version of the module. Used as cache buster for CDNs.
+ *                Note: the version can't be changed during runtime.
+ * @returns the status of the script/module
+ */
+export const useDynamicScript = (name: string, version: string): ScriptStatus => {
```

Comment:
I've separated the Kobs business logic and the script loading into two functions. Should make it easier to re-use it later